src/route-loading.ts

```diff
@@ -76,70 +100,77 @@ function determineWhatToLoad(
   return toLoad;
 }
 
-function loadRoute(
+async function loadRoute(
   routeLoader: RouteLoader,
   navigationInstruction: NavigationInstruction,
   viewPortPlan: ViewPortPlan
 ) {
-  let moduleId = viewPortPlan.config ? viewPortPlan.config.moduleId : null;
-
-  return loadComponent(routeLoader, navigationInstruction, viewPortPlan.config)
-    .then((component) => {
-      let viewPortInstruction = navigationInstruction.addViewPortInstruction(
-        viewPortPlan.name,
-        viewPortPlan.strategy,
-        moduleId,
-        component);
-
-      let childRouter = component.childRouter;
-      if (childRouter) {
-        let path = navigationInstruction.getWildcardPath();
-
-        return childRouter._createNavigationInstruction(path, navigationInstruction)
-          .then((childInstruction) => {
-            viewPortPlan.childNavigationInstruction = childInstruction;
-
-            return _buildNavigationPlan(childInstruction)
-              .then((childPlan) => {
-                if (childPlan instanceof Redirect) {
-                  return Promise.reject(childPlan);
-                }
-                childInstruction.plan = childPlan;
-                viewPortInstruction.childNavigationInstruction = childInstruction;
-
-                return loadNewRoute(routeLoader, childInstruction);
-              });
-          });
-      }
+  let config = viewPortPlan.config;
+  let moduleId: string | null = null;
+  let viewModel: ViewModelSpecifier | null = null;
+  if (config) {
+    if ('moduleId' in config) {
+      moduleId = config.moduleId;
```

Comment:
Small detail here. This will take any moduleId, even if it's `null`/`undefined`/`''`, and overwrite the existing `null` value with it. Are empty string and undefined accounted for as potential values now?

Reply:
It only comes when user defines it. So I'm just doing simple garbage in, garbage out here. In order to keep backward compat, `moduleId` > `viewModel`